test/trento/activity_logging/activity_logger_test.exs

```diff
@@ -95,6 +95,29 @@ defmodule Trento.ActivityLog.ActivityLoggerTest do
     end
   end
 
+  describe "user management activity detection" do
+    test "should trace user deletion", %{
+      conn: conn,
+      user: %{id: admin_id, username: admin_username}
+    } do
+      %{id: user_id, username: username} = insert(:user)
+
+      conn
+      |> with_token(admin_id)
+      |> delete("/api/v1/users/#{user_id}")
+
+      wait_for_tasks_completion()
+
+      assert [
+               %ActivityLog{
+                 type: "user_deletion",
+                 actor: ^admin_username,
+                 metadata: %{"username" => ^username, "user_id" => ^user_id}
```

Comment:
after deletion the username becomes `<real-username>__<deletion-timestamp>` so this is not going to match, unfortunately.

Reply:
I couldn't come up with a better option than checking string starts with :shrug:

Reply:
I see. We could consider making sure that when the `Users` context returns possibly deleted users we clean up the username.

Reply:
I also think that it's the `Users` context duty to come up with the cleanup logic

Reply:
Sweet, so are we going to address this here or in a follow up?

Reply:
I don't know :shrug:  

The only I know is it doesn't bother the activity log as it's something that refers to the user management logic.